standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1954,9 +2007,20 @@ private List<ColumnStatisticsObj> aggrStatsUseJava(String catName, String dbName
         areAllPartsFound, useDensityFunctionForNDVEstimation, ndvTuner);
   }
 
-  private List<ColumnStatisticsObj> aggrStatsUseDB(String catName, String dbName,
-      String tableName, List<String> partNames, List<String> colNames, String engine,
-      boolean areAllPartsFound, boolean useDensityFunctionForNDVEstimation, double ndvTuner) throws MetaException {
+  private Map<String, List<Object[]>> columnWiseSubList(List<Object[]> list) {
+    Map<String, List<Object[]>> colSubList = new HashMap<>();
+    for (Object[] row : list) {
+      String colName = (String) row[0];
+      colSubList.putIfAbsent(colName, new ArrayList<>());
```

Comment:
Instead of adding the elements to an `ArrayList` could you please aggregate the elements and store the aggregated data in the map.